prog/probe.go

```diff
@@ -148,7 +157,14 @@ func probeMain(flags probeFlags, targets []appclient.Target) {
 	}
 	defer resolver.Stop()
 
-	p := probe.New(flags.spyInterval, flags.publishInterval, clients, flags.noControls)
+	var publisher probe.ReportPublisher
+	if flags.printOnStdout {
+		publisher = new(stdoutPublisher)
+	} else {
+		publisher = clients
+	}
```

Comment:
This is quite hacky... the fact that we are still creating the client earlier on, even though we aren't publishing to it. I cannot think of an easy way around that though since various reporters need a `controls.PipeClient`. Perhaps we could feed in a noop implementation?